build/config/phpcompatibility.xml

```diff
@@ -0,0 +1,72 @@
+
+    <!--
+    #############################################################################
+    COMMAND LINE ARGUMENTS
+    https://github.com/squizlabs/PHP_CodeSniffer/wiki/Annotated-ruleset.xml
+    #############################################################################
+    -->
+
+    <file>./rector-prefixed-downgraded/</file>
+
+    <!-- Exclude PHPDoc structure files. -->
+    <exclude-pattern>*/rector-prefixed-downgraded/stubs/*</exclude-pattern>
+    <!-- these are never used, only required by symfony/http-kernel, but not really needed -->
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/symfony/error-handler/*</exclude-pattern>
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/symfony/var-dumper/*</exclude-pattern>
+
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/symfony/http-kernel/Resources/*</exclude-pattern>
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/rector/rector-nette/tests/*</exclude-pattern>
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/symfony/polyfill-mbstring/bootstrap80.php/*</exclude-pattern>
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/tracy/tracy/examples/*</exclude-pattern>
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/ssch/typo3-rector/templates/maker/*</exclude-pattern>
+    <exclude-pattern>*/rector-prefixed-downgraded/vendor/ssch/typo3-rector/stubs/*</exclude-pattern>
+
+
+    <rule name="PHPCompatibility.FunctionDeclarations.NewExceptionsFromToString.Found">
+        <!-- FALSE POSITIVE: wrapped in a PHP version check -->
+        <exclude-pattern>*/rector-prefixed-downgraded/vendor/nette/utils/src/Utils/Image.php</exclude-pattern>
+    </rule>
+
+    <rule name="PHPCompatibility.Keywords.ForbiddenNamesAsDeclared.stringFound">
+        <!-- namespace String token is allowed, not sure how to exclude that for T_NAMESPACE only, so is now global -->
+        <exclude-pattern>*/rector-prefixed-downgraded/vendor/symfony/string</exclude-pattern>
+    </rule>
+
+    <rule name="PHPCompatibility.Keywords.ForbiddenNamesAsDeclared.resourceFound">
+        <!-- namespace Resource token is allowed, not sure how to exclude that for T_NAMESPACE only, so is now global -->
+        <exclude-pattern>*/rector-prefixed-downgraded/vendor/symfony/config/Resource</exclude-pattern>
+    </rule>
+
+    <rule name="PHPCompatibility.FunctionUse.ArgumentFunctionsReportCurrentValue.NeedsInspection">
+        <!-- trusted source -->
+        <exclude-pattern>*/rector-prefixed-downgraded/vendor/nette/utils/src/Utils/Callback.php</exclude-pattern>
+        <exclude-pattern>*/rector-prefixed-downgraded/vendor/tracy/tracy/src/Tracy/Logger/FireLogger.php</exclude-pattern>
+        <exclude-pattern>*/rector-prefixed-downgraded/vendor/idiosyncratic/editorconfig/src/Section.php</exclude-pattern>
+    </rule>
```

Comment:
As a general best practice, I'd recommend, moving these down to _below_ the inclusion of the complete ruleset.
I personally normally put these in a section "Selective exclusions".

Also note: `exclude-pattern`s are basically PCRE regexes with automatic replacement of the `*` with `.*`, so patterns containing filenames are recommended to use `.../FileName\.php$` to prevent any confusion.

Reply:
Thanks. I'll tidy this up when error-trial part is over :) there is lot of false positive and I don't know how to ignore them correctly.

While you're here - how can ignore this rule for all `T_NAMESPACE` tokens?

```xml
<rule name="PHPCompatibility.Keywords.ForbiddenNamesAsDeclared.stringFound">
```

That would save me lot of manual ignoring.

> so note: exclude-patterns are basically PCRE regexes with automatic replacement of the * with .*, so patterns containing filenames are recommended to use .../FileName\.php$ to prevent any confusion.

Thanks! I'll update it

Reply:
> there is lot of false positive and I don't know how to ignore them correctly.

PHPCompatibility is set up to generate minimal noise, so if something is really a false positive, please check if there is an issue open about it and if not, report it and we can see if we can improve the scanner further.

As a general rule of thumb, warnings will normally refer to deprecations and things the scanner could not reliably determine, while errors are used for new and removed PHP features.

You could choose to ignore warnings when running the scan using `-n` or, alternatively, ignore warnings only for the exit code determination using `--runtime-set ignore_warnings_on_exit 1`.

Both of these can also be added to the ruleset:
* For `-n` adjust line 53 to read `<arg value="psn"/>`
* For `--runtime-set ignore_warnings_on_exit 1` add the following `<config name="ignore_warnings_on_exit" value="1"/>`


> While you're here - how can ignore this rule for all `T_NAMESPACE` tokens?

Unfortunately you can't. The errorcode (fourth part of the name) is modular, but based on the keyword found, not _where_ it was found.

>  how I can exclude single file without using regex? The regex is not really needed here.

Every `exclude-pattern` is passed through to a `preg_match()` call, so those are always regexes, whether you want it or not.

Reply:
I got a bit lost in this comment with too many new topics I don't need at the moment at once.
I understand that `preg_match()` is always used and namespace false positives cannot be exluded :+1: